idaes/generic_models/properties/core/eos/ideal.py

```diff
@@ -125,8 +161,24 @@ def energy_internal_mol_phase_comp(b, p, j):
 
     @staticmethod
     def enth_mol_phase(b, p):
-        return sum(b.get_mole_frac()[p, j]*b.enth_mol_phase_comp[p, j]
-                   for j in b.components_in_phase(p))
+        pobj = b.params.get_phase(p)
+        if pobj.is_vapor_phase():
+            return sum(b.get_mole_frac()[p, j]*b.enth_mol_phase_comp[p, j]
```

Comment:
Just a question on naming convention: Should this be modified at some point from `enth_mol_phase_comp[p, j]` to `enth_mol_vap_comp[j]`? 

I only thought this since `enth_mol_liq_comp` and `enth_mol_sol_comp` both exist.

Reply:
Actually, there appears to be an `enth_mol_ig_comp` for vapor phase. I don't think this would change the code's functionality but thought you might want to change for naming consistency.

Reply:
No, this is correct. It looks al little odd at first, but this is because the vapour phase does not have a pressure dependent term - thus `enth_mol_phase` is jus the sum of `enth_mol_phase_comp` across all components.

The other phases could do this too, but I thought the expressions would turn out neater if they were recalculated for the ideal properties (basically having `V*(P-Pref)` instead of `sum(x[i]*V*(P-Pref) for all i)`). Functionally it is the same, but I thought it best to avoid the unnecessary sum term.